presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -127,25 +125,14 @@ default ConnectorTableLayoutHandle getAlternativeLayoutHandle(ConnectorSession s
     }
 
     /**
-     * Experimental: if true, the engine will invoke pushdownFilter instead of getTableLayouts.
-     *
-     * This interface can be replaced with a connector optimizer rule once the engine supports these (#12546).
-     */
-    @Experimental
-    default boolean isPushdownFilterSupported(ConnectorSession session, ConnectorTableHandle tableHandle)
-    {
-        return false;
-    }
-
-    /**
-     * Experimental: returns table layout that encapsulates the given filter.
-     *
-     * This interface can be replaced with a connector optimizer rule once the engine supports these (#12546).
+     * Experimental: if true, the engine will invoke getLayout otherwise, getLayout will not be called.
+     * This function remains to ensure backwards compatibility, it needs to be removed.
      */
+    @Deprecated
     @Experimental
-    default ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, ConnectorTableHandle tableHandle, RowExpression filter, Optional<ConnectorTableLayoutHandle> currentLayoutHandle)
+    default boolean isPredicatePushdownEnabled(ConnectorSession session, ConnectorTableHandle tableHandle)
```

Comment:
let's rename to isLegacyGetLayoutSupported; that name would match the description and the actual meaning more closely